File: server/src/modules/auth/auth.controller.ts
import { Context } from "koa";
import authService from "./auth.service";
import { Result } from "@/core/result";

class AuthController {
  // 登录
  async login(ctx: Context) {
    const { userName, password } = ctx.request.body as any;
    const data = await authService.login(userName, password);
    ctx.body = Result.success(data, "登录成功");
  }

  // 注册
  async register(ctx: Context) {
    const data = ctx.request.body;
    const user = await authService.register(data);
    ctx.body = Result.success(user, "注册成功");
  }

  // 获取当前用户信息（含角色、权限），需要鉴权
  async info(ctx: Context) {
    const { id } = ctx.state.user;
    const data = await authService.getUserInfo(id);
    ctx.body = Result.success(data);
  }
}

export default new AuthController();

File: java-server/src/main/java/com/todothink/modules/auth/controller/AuthController.java
package com.todothink.modules.auth.controller;

import com.todothink.core.result.Result;
import com.todothink.core.security.SecurityUtils;
import com.todothink.modules.auth.dto.LoginRequest;
import com.todothink.modules.auth.dto.RegisterRequest;
import com.todothink.modules.auth.service.AuthService;
import org.springframework.validation.annotation.Validated;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.Map;

@RestController
@RequestMapping("/auth")
public class AuthController {

    private final AuthService authService;

    public AuthController(AuthService authService) {
        this.authService = authService;
    }

    @PostMapping("/login")
    public Result<Map<String, Object>> login(@Validated @RequestBody LoginRequest request) {
        Map<String, Object> data = authService.login(request.getUserName(), request.getPassword());
        return Result.success(data, "登录成功");
    }

    @PostMapping("/register")
    public Result<Object> register(@Validated @RequestBody RegisterRequest request) {
        return Result.success(authService.register(request), "注册成功");
    }

    @GetMapping("/info")
    public Result<Map<String, Object>> info() {
        Integer userId = SecurityUtils.currentUserId();
        return Result.success(authService.getUserInfo(userId));
    }
}
